src/commands/edit.rb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+
+#==============================================================================
+# Copyright (C) 2017 Stephen F. Norledge and Alces Software Ltd.
+#
+# This file/package is part of Alces Metalware.
+#
+# Alces Metalware is free software: you can redistribute it and/or
+# modify it under the terms of the GNU Affero General Public License
+# as published by the Free Software Foundation, either version 3 of
+# the License, or (at your option) any later version.
+#
+# Alces Metalware is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this package.  If not, see <http://www.gnu.org/licenses/>.
+#
+# For more information on the Alces Metalware, please visit:
+# https://github.com/alces-software/metalware
+#==============================================================================
+
+require 'system_command'
+require 'metal_log'
+
+module Metalware
+  module Commands
+    class Edit < CommandHelpers::BaseCommand
+      private
+
+      attr_reader :file
+
+      def setup
+        @file = args[0]
+      end
+
+      def run
+        cmd = "#{editor} #{file}"
+        MetalLog.info("exec: #{cmd}")
+        exec(cmd)
+      end
+
+      def editor
+        if !system_visual.empty?
+          system_visual
+        elsif !system_editor.empty?
+          system_editor
+        else
+          'vi'
+        end
+      end
+
+      def system_visual
+        @system_visual ||= SystemCommand.run('echo $VISUAL').chomp
```

Comment:
A simpler way to handle this and `system_editor` would be to use `ENV` I think, e.g. `ENV['VISUAL']`.